### DISCUSSION TITLE
Unable to use authentication

### DISCUSSION THREAD

**Discussion by NizarBlond** [rhubarbgroup/.github#4] [Support (ROC)]
The following are my wp-config.php settings.

I use Redis on localhost.

`
define( 'WP_REDIS_DATABASE', 3 );
define( 'WP_REDIS_PREFIX', 'NyTFl' );
define( 'WP_REDIS_PASSWORD', [ 'testuser', '123123' ]);
define( 'WP_REDIS_CLIENT', 'predis' );
define( 'WP_REDIS_SHARDS', false );
`

I've configured SHARED to false because for some reason in object-cache.php you only consider the password inside this loop:
`foreach ( [ 'WP_REDIS_SERVERS', 'WP_REDIS_SHARDS', 'WP_REDIS_CLUSTER' ] as $constant ) {
            if ( defined( $constant ) ) {
              ...`

and while I don't understand why, it is the only trick that allows me to use the auth feature.

However, I get this error:

Warning: strlen() expects parameter 1 to be string, array given in /var/www/sites/mysite/wp-content/plugins/redis-cache/dependencies/predis/predis/src/Connection/Factory.php on line 167

Fatal error: Uncaught Predis\Connection\ConnectionException: `SELECT` failed: NOAUTH Authentication required. [tcp://127.0.0.1:6379] in /var/www/sites/mysite/wp-content/plugins/redis-cache/dependencies/predis/predis/src/Connection/AbstractConnection.php:131

-

Any idea how to connect?

**Comment by tillkruss** [comment]
```
NOAUTH Authentication required
```

Read the FAQ.

**Reply by NizarBlond** [reply]
I read the FAQ, it states that it supported auth connections. I added:

define( 'WP_REDIS_PASSWORD', [ 'testuser', '123123' ]); 

But I don't get it to work.

Ideas?

**Reply by tillkruss** [reply]
If you followed the FAQ, then try flushing your OPcode cache. Your `wp-config.php` might be cached by PHP.
